contrib/flo-styx/src/main/java/com/spotify/flo/contrib/styx/LabelUtil.java

```diff
@@ -0,0 +1,133 @@
+        STYX_COMPONENT_ID_LABEL_KEY);
+    final String workflowIdLabelKey = validateAndPrefixLabelKey(labelPrefix,
+        STYX_WORKFLOW_ID_LABEL_KEY);
+    final String parameterLabelKey = validateAndPrefixLabelKey(labelPrefix,
+        STYX_PARAMETER_LABEL_KEY);
+    final String executionIdLabelKey = validateAndPrefixLabelKey(labelPrefix,
+        STYX_EXECUTION_ID_LABEL_KEY);
+    final String triggerIdLabelKey = validateAndPrefixLabelKey(labelPrefix,
+        STYX_TRIGGER_ID_LABEL_KEY);
+    final String triggerTypeLabelKey = validateAndPrefixLabelKey(labelPrefix,
+        STYX_TRIGGER_TYPE_LABEL_KEY);
+    
+    sanitizeLabelValue("c", STYX_COMPONENT_ID, config)
+        .ifPresent(value -> map.put(componentIdLabelKey, value));
+    sanitizeLabelValue("w", STYX_WORKFLOW_ID, config)
+        .ifPresent(value -> map.put(workflowIdLabelKey, value));
+    sanitizeLabelValue("p", STYX_PARAMETER, config)
+        .ifPresent(value -> map.put(parameterLabelKey, value));
+    sanitizeLabelValue("e", STYX_EXECUTION_ID, config)
+        .ifPresent(value -> map.put(executionIdLabelKey, value));
+    sanitizeLabelValue("t", STYX_TRIGGER_ID, config)
+        .ifPresent(value -> map.put(triggerIdLabelKey, value));
+    sanitizeLabelValue("tt", STYX_TRIGGER_TYPE, config)
+        .ifPresent(value -> map.put(triggerTypeLabelKey, value));
+
+    return map;
+  }
+
+  private static String validateAndPrefixLabelKey(String labelPrefix, String label) {
+    final String prefixedLabelKey = String.format("%s-%s", labelPrefix, label);
+
+    if (prefixedLabelKey.length() > LABEL_KEY_MAX_LENGTH) {
+      throw new IllegalArgumentException("Invalid label key: Too long, must be <= 63 characters: "
+                                         + prefixedLabelKey);
+    }
+    
+    if (!prefixedLabelKey.matches(LABEL_KEY_REGEX)) {
+      throw new IllegalArgumentException("Invalid label key: " + prefixedLabelKey);
+    }
+    
+    return prefixedLabelKey;
+  }
+
+  private static Optional<String> sanitizeLabelValue(String pad, String path, Config config) {
+    if (!config.hasPath(path)) {
+      return Optional.empty();
+    }
+
+    // Only allow [-a-z0-9]
+    String value = config.getString(path).toLowerCase().replaceAll(LABEL_VALUE_REPLACE_REGEX, "-");
+
+    // First character must be [a-z]
+    if (!Character.isAlphabetic(value.charAt(0))) {
+      value = String.format("%s-%s", pad, value);
+    }
+
+    // We always assume padding at the end is needed to be on the safe side
+    // so the worst case is we trim more than we actually need
+    if (value.length() + pad.length() > LABEL_VALUE_MAX_LENGTH) {
+      value = value.substring(0, LABEL_VALUE_MAX_LENGTH - pad.length());
+    }
+
+    // Last character must be [a-z0-9] (not -)
+    if (value.charAt(value.length() - 1) == '-') {
+      value = String.format("%s%s", value, pad);
+    }
+
+    // Final check
+    if (!value.matches(LABEL_VALUE_REGEX)) {
```

Comment:
This is pretty much being paranoid I think. Can't find a case.

Reply:
Seems like it, can't think of a case for it either